add docs for derive topn from window (#12946)
pingcap/docs: system-variables.md

- Default value: `3.0`
- Indicates the CPU cost for TiDB to process one row. This variable is internally used in the [Cost Model](/cost-model.md), and it is **NOT** recommended to modify its value.

### `tidb_opt_derive_topn` <span class="version-mark">New in v7.0.0</span>

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Type: Boolean
- Default value: `OFF`

> - This variable is used to set whether to enable the optimization of [Deriving TopN or Limit from window functions](/derive-topn-from-window.md).

Reviewer: ```suggestion
- Controls whether to enable the optimization rule of [Deriving TopN or Limit from window functions](/derive-topn-from-window.md).
```
